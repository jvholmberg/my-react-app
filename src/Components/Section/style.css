.Section {
  display: block;
  padding: 5rem 0;
  width: 100%;
  box-sizing: border-box;
  overflow: hidden;
}


/**
 * Appearances
 */
.Section-Normal {
  color: #000;
}
.Section-Inverted {
  background-color: #343a40;
  color: #fff;
}


/**
 * Alignment
 */
.Section-Left-Aligned {
  text-align: left;
}
.Section-Center-Aligned {
 text-align: center;
}
.Section-Right-Aligned {
 text-align: right;
}
