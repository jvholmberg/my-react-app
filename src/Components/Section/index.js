import React from 'react';
import { CraftClassName } from '../';

import './style.css';

/**
 * A Section of your page presenting content of type X.
 * @param {boolean} inverted - Adjust content to be viewable on dark background
 * @param {boolean} left - Make content horizontally left aligned
 * @param {boolean} center - Make content horizontally center aligned
 * @param {boolean} right - Make content horizontally right aligned
 * @returns {element} <Section>
 */
const Section = ({ inverted, left, center, right, className, children, ...props }) => {
  let cn = CraftClassName({ inverted }, 'Section', 'Normal', 'Section-');
  cn = CraftClassName({ left, center, right }, cn, 'Left', 'Section-', '-Aligned');
  if (className) cn = `${className} ${cn}`;
  return (
    <div className={cn} {...props}>
      {children}
    </div>
  )
};
export default Section;
