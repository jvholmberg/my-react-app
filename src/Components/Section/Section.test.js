import React from 'react';

import Section from './';

describe('Section', () => {
  it('Empty matches snapshot', () => {
    expect(render(
      <Section />
    )).toMatchSnapshot();
  });
  it('With className matches snapshot', () => {
    expect(render(
      <Section className='className' />
    )).toMatchSnapshot();
  });
  it('Left-aligned with className matches snapshot', () => {
    expect(render(
      <Section className='className' left />
    )).toMatchSnapshot();
  });
});
