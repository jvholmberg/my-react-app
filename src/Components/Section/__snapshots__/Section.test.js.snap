// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Empty matches snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "Section Section-Normal Section-Left-Aligned",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Section Left-aligned with className matches snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "className Section Section-Normal Section-Left-Aligned",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Section With className matches snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "className Section Section-Normal Section-Left-Aligned",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
